pallets/build-evm-tx/src/lib.rs

```diff
@@ -0,0 +1,115 @@
+// This file is part of HydraDX.
+
+// Copyright (C) 2020-2021  Intergalactic, Limited (GIB).
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::pallet_prelude::*;
+use frame_system::{ensure_signed, pallet_prelude::OriginFor};
+use sp_std::vec::Vec;
+
+use ethereum::{AccessListItem, EIP1559TransactionMessage, TransactionAction};
+use sp_core::{H160, U256};
+
+pub use pallet::*;
+
+const EIP1559_TX_TYPE: u8 = 0x02;
+
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Maximum length of transaction data
+		#[pallet::constant]
+		type MaxDataLength: Get<u32>;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Transaction data exceeds maximum allowed length
+		DataTooLong,
+		/// Invalid address format - must be exactly 20 bytes
+		InvalidAddress,
+		/// Priority fee cannot exceed max fee per gas (EIP-1559 requirement)
+		InvalidGasPrice,
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Build an EIP-1559 EVM transaction and return the RLP-encoded data
+		///
+		/// # Parameters
+		/// - `origin`: The signed origin
+		/// - `to_address`: Optional recipient address (None for contract creation)
+		/// - `value`: ETH value in wei
+		/// - `data`: Transaction data/calldata
+		/// - `nonce`: Transaction nonce
+		/// - `gas_limit`: Maximum gas units for transaction
+		/// - `max_fee_per_gas`: Maximum total fee per gas (base + priority)
+		/// - `max_priority_fee_per_gas`: Maximum priority fee (tip) per gas
+		/// - `chain_id`: Target EVM chain ID
+		///
+		/// # Returns
+		/// RLP-encoded transaction data with EIP-2718 type prefix (0x02 for EIP-1559)
+		pub fn build_evm_tx(
```

Comment:
it feels like this does not need to be a separate pallet. 

i believe it can be part of the signet pallet. so we have all in one place.